### PR TITLE
CLOUDP-114862: [Atlas CLI / MongoCLI] Improve error handling when user is unathenticated

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-114862

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

This PR updates the missing credential error message to include `auth login`

**Before**: 
```
./bin/mongocli atlas cluster ls
Error: missing credentials

To set credentials, run: mongocli config
```

**After**:
```
./bin/atlas cluster ls
Error: this actions requires authentication

To login using your Atlas username and password, run: atlas auth login
To set credentials using API keys, run: atlas config init
```

```
./bin/mongocli atlas cluster ls
Error: this actions requires authentication

To login using your Atlas username and password, run: mongocli auth login
To set credentials using API keys, run: mongocli config
```

### CODE REVIEW

File: internal/validate/validate.go [hunk 1]
```diff
@@ -93,7 +93,7 @@ func ObjectID(s string) error {
 	return nil
 }
 
-var ErrMissingCredentials = errors.New("missing credentials")
+var ErrMissingCredentials = errors.New("this actions requires authentication")
```
Comment (gssbzn) [inline internal/validate/validate.go:96]: this action or these actions
```suggestion
var ErrMissingCredentials = errors.New("this action requires authentication")
```

**Review by gssbzn — DISMISSED** [review]
LGTM if LGTDOCS the previous copy was approved by them

**Review by johansteyn — APPROVED** [review]
LGTM

File: internal/validate/validate.go [hunk 2]
```diff
@@ -109,9 +109,10 @@ func Credentials() error {
 		configCMD += " init"
 	}
 	return fmt.Errorf(
-		"%w\n\nTo set credentials, run: %s %s",
+		"%w\n\nTo login using your Atlas username and password, run: %s auth login\nTo set credentials using API keys, run: %s %s",
```
Comment (sarahsimpers) [inline internal/validate/validate.go:112]: "log in" when used as a verb vs "login" as a noun :) thanks!
```suggestion
		"%w\n\nTo log in using your Atlas username and password, run: %s auth login\nTo set credentials using API keys, run: %s %s",
```

**Review by sarahsimpers — APPROVED** [review]
LGTM % small change

File: internal/validate/validate.go (hunk 2 above)
Comment (andreaangiolillo) [reply internal/validate/validate.go]: updated

**Review by blva — APPROVED** [review]

**Pull request merged**